### PR TITLE
Feature/registration arch update

### PR DESCRIPTION
**Pull request by mateeullahmalik**
https://pastel.wiki/en/Architecture/Workflows/NewArtRegistration
https://pastel.wiki/en/Architecture/Components/TicketStructures
https://pastel-network.atlassian.net/browse/PSL-140

### CODE REVIEW

File: walletnode/services/artworkregister/task.go [hunk 1]
```diff
@@ -716,38 +733,83 @@ func (task *Task) connectToTopRankNodes(ctx context.Context) error {
...
 	}
+
 	task.fingerprintAndScores = task.nodes.FingerAndScores()
 	task.UpdateStatus(StatusImageProbed)
 
 	// As we are going to store the the compressed figerprint to kamedila
-	// so we calculated the hash base on the compressed fingerprint print also
-	fingerprintsHash, err := sha3256hash(task.fingerprintAndScores.ZstdCompressedFingerprint)
+	// so we calculated the hash based on the compressed fingerprints as well
+	fingerprintsHash, err := utils.Sha3256hash(task.fingerprintAndScores.ZstdCompressedFingerprint)
```
Comment (nguyen202) [inline walletnode/services/artworkregister/task.go:807]: `task.fingerprintAndScores = task.nodes.FingerAndScores()`
The fingerprintAndScores.ZstdCompressedFingerprint was not set in task.nodes.FingerAndScores() - should set it.

File: walletnode/services/artworkregister/task.go [hunk 2]
```diff
@@ -198,6 +218,40 @@ func (task *Task) run(ctx context.Context) error {
...
+	task.ddAndFingerprintsIDs, _, err = pastel.GetIDFiles(ddFpFile, task.ddAndFingerprintsIc, task.config.DDAndFingerprintsMax)
+	if err != nil {
+		return fmt.Errorf("get ID Files: %w", err)
+	}
+
+	comp, err := zstd.CompressLevel(nil, ddFpFile, 22)
+	if err != nil {
+		return errors.Errorf("compress: %w", err)
+	}
+	task.ddAndFpFile = utils.B64Encode(comp)
```
Comment (nguyen202) [inline walletnode/services/artworkregister/task.go:250]: while we need to encode compress to base64?

File: proto/walletnode/register_artwork.proto [hunk 3]
```diff
@@ -125,8 +97,10 @@ message SendSignedNFTTicketRequest {
     string key1 = 3;
     string key2 = 4;
     EncoderParameters encode_parameters = 5;
-    map<string, bytes> encode_files = 6;
+    bytes dd_fp_files = 7;
```
Comment (nguyen202) [inline proto/walletnode/register_artwork.proto:100]: should be :
    bytes dd_fp_files = **6;**
    bytes rq_files = **7;**

File: supernode/services/artworkregister/task.go [hunk 4]
```diff
@@ -378,6 +638,12 @@ func (task *Task) ValidatePreBurnTransaction(ctx context.Context, txid string) (
 						return nil
 					}
 
+					if err = task.storeIDFiles(ctx); err != nil {
```
Comment (nguyen202) [inline supernode/services/artworkregister/task.go:641]: I don't think storeFingerprints() in line 635 is necessary any more?

File: walletnode/services/artworkregister/task.go (hunk 2 above)
Comment (mateeullahmalik) [reply walletnode/services/artworkregister/task.go:250]: @nguyen202 please read step 11A

File: proto/walletnode/register_artwork.proto (hunk 3 above)
Comment (mateeullahmalik) [reply proto/walletnode/register_artwork.proto:100]: ok

File: supernode/services/artworkregister/task.go (hunk 4 above)
Comment (mateeullahmalik) [reply supernode/services/artworkregister/task.go:641]: i'm not sure

File: walletnode/services/artworkregister/task.go (hunk 1 above)
Comment (mateeullahmalik) [reply walletnode/services/artworkregister/task.go:807]: @nguyen202 not sure what you mean, sry

File: supernode/services/artworkregister/task.go (hunk 4 above)
Comment (mateeullahmalik) [reply supernode/services/artworkregister/task.go:641]: ok i've removed it after confirmation

File: walletnode/services/artworkregister/task.go (hunk 1 above)
Comment (nguyen202) [reply walletnode/services/artworkregister/task.go:807]: I did not see any where to set `ZstdCompressedFingerprint`, so we have null value here.

File: walletnode/services/artworkregister/task.go (hunk 2 above)
Comment (nguyen202) [reply walletnode/services/artworkregister/task.go:250]: Thanks.

File: walletnode/services/artworkregister/task.go (hunk 1 above)
Comment (mateeullahmalik) [reply walletnode/services/artworkregister/task.go]: @nguyen202 pls push the fix for this?

**Review by a-ok123 — APPROVED** [review]

**Review by Dicklesworthstone — APPROVED** [review]

**Pull request merged**